Improve dev process (#6057)
NethServer/dev: developer-manual/development_process.rst

By the way, it’s perfectly reasonable not to fill issues for
occasional small fixes, like typos in translations.

When implementing small fixes, always avoid commits to the master branch.
Open a :ref:`pull_request-section` and carefully describe the problem.
Creation of issues can be avoided only for trivial fixes which require

> minimal QA effort.

Reviewer: I'd be stricter. If I give it to QA people for testing it is not trivial. 

```suggestion
no QA effort.
```

* Finally, clears the *Assignee*.

> If the issue is not valid, it can be closed using the **invalid** label

Reviewer: ```suggestion
If the issue is not valid, it must be closed using the **invalid** label.
```

> and adding the reason inside the description like *duplicate* or *wontfix*.

Reviewer: ```suggestion
A comment must convey the reason why it is invalid, like *duplicate of (URL of issue)*, *wontfix because ...".
```

Author: Because I want to know who is working on a draft, otherwise I will fell free to close it :evi

    NethServer/dev#1133

Draft pull requests
-------------------

The use of draft pull requests is recommended to share an on-going development.
Draft pull requests can be used to test possible implementations of features that do not have an issue yet.

> A draft pull request should always have an assignee.

Reviewer: Why?

```suggestion
```

Author: Because I want to know who is working on a draft, otherwise I will fell free to close it :smiling_imp:

Reviewer: Isn't the draft author enough (by default)?

Author: Usually not, because some users open the PR and then forgot it for months hoping that someone will jump in :)

If nobody is assignee, nobody is working it: just like the issue.

Reviewer: Ok now I understand. In this case this is my proposal:

```suggestion
If the draft pull request does not reference an issue it must have an assignee.
```

That should be true also for normal PRs that do not reference any issue :thinking:
